ocs/cluster.py

```diff
@@ -0,0 +1,308 @@
+
+This module has rook related classes, support for functionalities to work with
+rook cluster. This works with assumptions that an OCP cluster is already
+functional and proper configurations are made for interaction.
+"""
+
+import logging
+import base64
+
+import resources.pod as pod
+from resources import ocs
+import ocs.defaults as default
+from utility.utils import TimeoutSampler
+from ocsci.config import ENV_DATA
+from ocs import ocp
+from ocs import exceptions
+
+POD = ocp.OCP(kind='Pod', namespace=ENV_DATA['cluster_namespace'])
+CEPHCLUSTER = ocp.OCP(
+    kind='CephCluster', namespace=ENV_DATA['cluster_namespace']
+)
+CEPHFS = ocp.OCP(
+    kind='CephFilesystem', namespace=ENV_DATA['cluster_namespace']
+)
+
+logger = logging.getLogger(__name__)
+
+
+class CephCluster(object):
+    """
+    Handles all cluster related operations from ceph perspective
+
+    This class has depiction of ceph cluster. Contains references to
+    pod objects which represents ceph cluster entities.
+
+    Attributes:
+        _ceph_pods (list) : A list of  ceph cluster related pods
+        _cluster_name (str): Name of ceph cluster
+        _namespace (str): openshift Namespace where this cluster lives
+    """
+
+    def __init__(self):
+        """
+        Cluster object initializer, this object needs to be initialized
+        after cluster deployment. However its harmless to do anywhere.
+        """
+        # cluster_name is name of cluster in rook of type CephCluster
+
+        self.cluster_resource_config = CEPHCLUSTER.get().get('items')[0]
+        try:
+            self.cephfs_config = CEPHFS.get().get('items')[0]
+        except IndexError as e:
+            logging.warning(e)
+            logging.warning("No CephFS found")
+            self.cephfs_config = None
+
+        self._cluster_name = self.cluster_resource_config.get(
+            'metadata').get('name')
+        self._namespace = self.cluster_resource_config.get(
+            'metadata').get('namespace')
+
+        # We are not invoking ocs.create() here
+        # assuming cluster creation is done somewhere after deployment
+        # So just load ocs with existing cluster details
+        self.cluster = ocs.OCS(**self.cluster_resource_config)
+        if self.cephfs_config:
+            self.cephfs = ocs.OCS(**self.cephfs_config)
+
+        self.mon_selector = default.MON_APP_LABEL
+        self.mds_selector = default.MDS_APP_LABEL
+        self.tool_selector = default.TOOL_APP_LABEL
+        self.mons = []
+        self._ceph_pods = []
+        self.mdss = []
+        self.toolbox = None
+        self.mds_count = 0
+        self.mon_count = 0
+
+        self.scan_cluster()
+
+        # Set counts after scan_cluster to get latest updates on pod counts
+        self.mon_count = len(self.mons)
+        if self.cephfs_config:
+            if self.cephfs.data['spec']['metadataServer']['activeStandby']:
+                self.mds_count = int(len(self.mdss) / 2)
```

Comment:
if you have activeStandby, it should be * instead of /
Right? cause for every mds, you will have 1 more as standby

Reply:
Here I thought of providing only the activeCount to the user because only other reference we have to match against is the activeCount parameter from yaml (OR through OCS.get()) . If we store total number of mdss here then user has to do calculations on what should be the active count.  If any of the standby pods are down then definitely we can catch in mds_health().  Also one standby each need not be always true , standby works based on ranks of active mdss, for ex: A single mds may have 2 standby so instead of offloading all these calculation burden to tests I felt its better to have it here in library. 

In any case let me know if this is confusing to the user, I will change.

Reply:
Changed mds_count as absolute count of mds pods and user has to do necessary arithmetic in tests to find active count